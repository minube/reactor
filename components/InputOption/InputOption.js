import { bool, func, string } from 'prop-types';
import React, { Fragment } from 'react';
import { View } from 'react-native';

import Icon from '../Icon';
import InputLabel from '../Input/InputLabel';
import Motion from '../Motion';
import Touchable from '../Touchable';
import styles from './InputOption.style';

const InputOption = ({
  label, onChange, rounded, value, ...inherit
}) => (
  <Touchable onPress={onChange ? () => onChange(!value) : undefined} style={[styles.container, inherit.style]}>
    <Fragment>
      <View style={[styles.option, rounded && styles.rounded, value && styles.selected]}>
        <Motion preset="pop" visible={value}>
          <Icon value="doneContrast" reverse style={styles.icon} />
        </Motion>
      </View>
      { label && (
        <InputLabel style={styles.label}>
          {label}
        </InputLabel>
      )}
    </Fragment>
  </Touchable>
);

InputOption.propTypes = {
  label: string,
  onChange: func,
  rounded: bool,
  value: bool,
};

InputOption.defaultProps = {
  label: undefined,
  onChange: undefined,
  rounded: true,
  value: false,
};

export default InputOption;
