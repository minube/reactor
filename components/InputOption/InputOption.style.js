import { Platform, StyleSheet } from 'react-native';

import { LAYOUT, THEME } from '../../common';

const { COLOR, UNIT } = THEME;

const SIZE = UNIT * 2;

export default StyleSheet.create({
  container: LAYOUT.STYLE.ROW,

  icon: {
    height: SIZE * 0.6,
    width: SIZE * 0.6,
  },

  label: {
    marginLeft: UNIT,
    ...Platform.select({
      web: {
        userSelect: 'none',
      },
    }),
  },

  option: {
    ...LAYOUT.STYLE.CENTERED,
    height: SIZE,
    width: SIZE,

    borderWidth: 2,
    borderColor: COLOR.TEXT_LIGHTEN,
    borderRadius: SIZE / 10,
  },

  rounded: {
    borderRadius: SIZE / 2,
  },

  selected: {
    borderColor: COLOR.PRIMARY,
    backgroundColor: COLOR.PRIMARY,
  },
});
