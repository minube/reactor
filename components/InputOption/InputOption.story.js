import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { boolean, object, text } from '@storybook/addon-knobs/react';

import InputOption from './InputOption';

const style = {
  backgroundColor: 'rgba(0,255,0,0.25)',
  padding: 10,
};

const LABEL = 'Lorem Ipsum...';

storiesOf('✅ InputOption', module)
  .add('default', () => (
    <InputOption />
  ))
  .add('label', () => (
    <InputOption label={LABEL} />
  ))
  .add('selected', () => (
    <InputOption selected />
  ))
  .add('rounded (false)', () => (
    <InputOption rounded={false} />
  ))
  .add('inherit:style', () => (
    <InputOption style={style} />
  ))
  .add('⚡ onChange', () => (
    <InputOption onChange={action('InputOption.onChange()')} />
  ))
  .add('🏀 Playground', () => (
    <InputOption
      label={text('label', LABEL)}
      selected={boolean('selected', true)}
      rounded={boolean('rounded', true)}
      onChange={action('InputOption.onChange()')}
      style={object('style', style)}
    />
  ));
