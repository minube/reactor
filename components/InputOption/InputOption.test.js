import React from 'react';
import renderer from 'react-test-renderer';

import InputOption from './InputOption';

const style = {
  backgroundColor: 'rgba(0,255,0,0.25)',
  padding: 10,
};
const LABEL = 'Lorem Ipsum...';

describe('<InputOption>', () => {
  it('renders', () => {
    const tree = renderer.create(<InputOption />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('label', () => {
    const tree = renderer.create(<InputOption label={LABEL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('value', () => {
    const tree = renderer.create(<InputOption value />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('rounded (false)', () => {
    const tree = renderer.create(<InputOption rounded={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('inherity:style', () => {
    const tree = renderer.create(<InputOption style={style} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('onChange', () => {
    // @TODO: Test <InputOption /> {onChange}
  });
});
