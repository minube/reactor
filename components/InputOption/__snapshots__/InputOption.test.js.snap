// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputOption> inherity:style 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 20,
      },
      Object {
        "backgroundColor": "rgba(0,255,0,0.25)",
        "padding": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "borderColor": "#707070",
          "borderRadius": 2,
          "borderWidth": 2,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "borderRadius": 10,
        },
        false,
      ]
    }
  >
    <View
      style={
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../components/Icon/assets/done_contrast.png",
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 20,
              "width": 20,
            },
            Object {
              "height": 14,
              "width": 14,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#cdd2d4",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<InputOption> label 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "borderColor": "#707070",
          "borderRadius": 2,
          "borderWidth": 2,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "borderRadius": 10,
        },
        false,
      ]
    }
  >
    <View
      style={
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../components/Icon/assets/done_contrast.png",
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 20,
              "width": 20,
            },
            Object {
              "height": 14,
              "width": 14,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#4a4a4a",
          "fontFamily": undefined,
          "fontWeight": "400",
          "textAlignVertical": "center",
        },
        Object {
          "fontFamily": undefined,
          "fontSize": 14,
          "fontWeight": "600",
          "lineHeight": 16,
        },
        Object {
          "color": "#707070",
        },
        Object {
          "marginBottom": 0,
          "marginLeft": 10,
        },
      ]
    }
  >
    Lorem Ipsum...
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#cdd2d4",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<InputOption> renders 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "borderColor": "#707070",
          "borderRadius": 2,
          "borderWidth": 2,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "borderRadius": 10,
        },
        false,
      ]
    }
  >
    <View
      style={
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../components/Icon/assets/done_contrast.png",
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 20,
              "width": 20,
            },
            Object {
              "height": 14,
              "width": 14,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#cdd2d4",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<InputOption> rounded (false) 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "borderColor": "#707070",
          "borderRadius": 2,
          "borderWidth": 2,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        false,
        false,
      ]
    }
  >
    <View
      style={
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../components/Icon/assets/done_contrast.png",
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 20,
              "width": 20,
            },
            Object {
              "height": 14,
              "width": 14,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#cdd2d4",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`<InputOption> value 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "borderColor": "#707070",
          "borderRadius": 2,
          "borderWidth": 2,
          "height": 20,
          "justifyContent": "center",
          "width": 20,
        },
        Object {
          "borderRadius": 10,
        },
        Object {
          "backgroundColor": "#10c0bc",
          "borderColor": "#10c0bc",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "opacity": 1,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../components/Icon/assets/done_contrast.png",
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "height": 20,
              "width": 20,
            },
            Object {
              "height": 14,
              "width": 14,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#cdd2d4",
          "height": 0,
          "opacity": 0,
          "width": 0,
          "zIndex": 0,
        }
      }
    />
  </View>
</View>
`;
